index.js

```diff
@@ -1,17 +1,57 @@
 'use strict';
 
-var isCallable = require('is-callable');
-var fnToStr = Function.prototype.toString;
-var isNonArrowFnRegex = /^\s*function/;
-var isArrowFnWithParensRegex = /^\([^)]*\) *=>/;
-var isArrowFnWithoutParensRegex = /^[^=]*=>/;
+/** @param {number} n */
+var found = function (n) {
+	return n !== -1;
+};
+/** @param {number} n */
+var notFoundGoesLast = function (n) {
+	return found(n) ? n : Infinity;
+};
 
+/** @param {unknown} fn */
 module.exports = function isArrowFunction(fn) {
-	if (!isCallable(fn)) {
+	if (typeof fn !== 'function') {
+		return false;
+	}
+
+	/** @type {string} */
+	var fnStr = Function.prototype.toString.call(fn);
```

Comment:
caching this at module level is strictly necessary to avoid someone altering this property.

Reply:
Couldn't they alter it before the module is imported though? I guess caching is still somewhat safer though

Reply:
Yes, there's no defense against code that runs before you - but caching at module-level is a defense against code that runs after you.